numpy/linalg/linalg.py

```diff
@@ -1659,7 +1663,8 @@ def pinv(a, rcond=1e-15 ):
 
     """
     a, wrap = _makearray(a)
-    _assertNoEmpty2d(a)
+    if product(a.shape[-2:]) == 0:
+        return wrap(empty(a.shape[:-2] + (a.shape[-1], a.shape[-2]), dtype=a.dtype))
```

Comment:
Line too long. I'd split out the shape computation into a new line or use
`a:shape[:-2] + a.shape[-1:-3:-1]`.